Rubberduck.Core/UI/CodeExplorer/Commands/ImportCommand.cs

```diff
@@ -86,9 +89,53 @@ protected override void OnExecute(object parameter)
 
             foreach (var filename in _openFileDialog.FileNames)
             {
-                using (var components = project.VBComponents)
+                var extension = filename.Split('.').Last();
+
+                var sourceText = string.Join(Environment.NewLine, File.ReadAllLines(filename));
+
+                string updatedModuleText;
+
+                var result = VBACodeStringParser.Parse(sourceText, t => t.startRule());
+
+                var tempHelper = (CodeExplorerItemViewModel)parameter;
+                var updatedFolder = (parameter is CodeExplorerCustomFolderViewModel) ? tempHelper.Name : tempHelper.GetSelectedDeclaration().CustomFolder;
+
+                const string folderAnnotation = "'@Folder";
+                if (result.parseTree.GetChild(0).GetText().Contains(folderAnnotation))
+                {
+                    var workingTree = TreeContainingFolderAnnotation(result.parseTree, folderAnnotation);
+
+                    var originalAnnotation = workingTree.GetText();
+
+                    var originalFolder = FolderNameFromFolderAnnotation(workingTree, folderAnnotation);
+
+                    var updatedFolderAnnotation = originalAnnotation.Replace(originalFolder, updatedFolder);
+                    result.rewriter.Replace(workingTree.SourceInterval.a, workingTree.SourceInterval.b, updatedFolderAnnotation);
+
+                    updatedModuleText = result.rewriter.GetText();
+                }
+                else
                 {
-                    components.Import(filename);
+                    updatedModuleText = $"{folderAnnotation}({updatedFolder}){Environment.NewLine}" + result.rewriter.GetText();
+                }
+
+                try
+                {
+                    var tempFile = $"RubberduckTempImportFile.{extension}";
+                    var sw = File.CreateText(tempFile);
+                    sw.Write(updatedModuleText);
+                    sw.Close();
+
+                    using (var components = project.VBComponents)
+                    {
+                        components.Import(tempFile);
+                    }
+
+                    File.Delete(tempFile);
+                }
+                catch
+                {
+                    Logger.Log(LogLevel.Error, "Unable to create temporary file to import into the correct folder while expecuting " + nameof(ImportCommand));
```

Comment:
make that `catch(Exception e)`, and log the exception details with `Logger.Error(e)`.

Reply:
in more than one place we just log a warning/error and log the execption at trace/debug to keep the log cleaner.

Also: typo